types/src/timestamp.rs

```diff
@@ -174,6 +201,7 @@ impl<'de> Deserialize<'de> for Timestamp {
     }
 }
 
+#[cfg(any(feature = "std", test))]
```

Comment:
Do we need the bytesrepr impls for both the time types to be feature-gated?

Reply:
IIRC, the main issue was that bytesrepr serialization returns a `Vec` - but maybe this can be worked around with `core::alloc`? I can try removing these feature gates and see if I can do so.